crowbar_framework/spec/lib/crowbar/lock/local_blocking_spec.rb

```diff
@@ -0,0 +1,66 @@
+require "spec_helper"
+require "timeout"
+
+describe Crowbar::Lock::LocalBlocking do
+  let(:lock) { subject.class.new }
+
+  after(:each) do
+    lock.release
+  end
+
+  context "when a lock is acquired" do
+    it "returns a lock object" do
+      expect(lock.acquire).to be_an_instance_of(subject.class)
+    end
+
+    it "sets a lock object to locked" do
+      expect(lock.locked?).to be false
+      lock.acquire
+      expect(lock.locked?).to be true
+    end
+
+    it "works via a #with_lock block" do
+      expect(lock.locked?).to be false
+      lock.with_lock do
+        expect(lock.locked?).to be true
+      end
+      expect(lock.locked?).to be false
+    end
+
+    it "is an exclusive lock" do
+      # Check that one of two attempts racing to obtain lock will win
+      lock2 = subject.class.new
+      lock.acquire
+      expect {
```

Comment:
Could do, yes.  In general I agree with this guideline though - `{}` for single-line blocks and `do ... end` for multi-line.  I think this is just a rare exception to that guideline - but that's why it's called "style guide" not "style rules" ;-)